### PR TITLE
Clean up resources on close. Allow clean application termination.

### PR DESCRIPTION
**Pull request by tszmytka**
An issue submitted in the former repository #1203 has not been identified correctly (at least partly). 
There still is a bug where cleanly closing your application by invoking `org.springframework.context.ConfigurableApplicationContext#close` is not possible if an `ExecutorService` has been initiated in `org.springframework.cloud.stream.binder.kafka.KafkaBinderMetrics#bindTo`. Other Spring resources get closed but the scheduler remains hanging and keeps the application JVM process running.
This PR fixes that and allows proper termination.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -12,3 +12,7 @@ insert_final_newline = true
 [*.yml]
 indent_style = space
 indent_size = 2
+
+[*.java]
+ij_java_names_count_to_use_import_on_demand = 999
+ij_java_class_count_to_use_import_on_demand = 999
```
Comment (sobychacko) [inline .editorconfig:18]: Why is this change necessary?

File: binders/kafka-binder/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderMetrics.java [hunk 2]
```diff
@@ -64,7 +65,7 @@
  * @author Lars Bilger
  */
```
Comment (sobychacko) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderMetrics.java:66]: Please add your name as an author. Also, make sure that any Java classes modified have the updated copyright years. Thanks!

**Review by sobychacko — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 1 above)
Comment (tszmytka) [reply .editorconfig:18]: As the commit message says: this instructs the IDE to not try to bundle multiple imports into one with a star (default behavior in Intellij IDEA) - this seems to be the repo's preference. Admittedly it has no direct connection with the feature but keeps you from messing up when you use automatic code formatting.

Comment (sobychacko) [reply .editorconfig:18]: That is fine. The checkstyle rules would prevent using asterisks for multiple imports.

**Comment by sobychacko** [comment]
Thanks for the PR addressing this issue. I squashed your commits, polished and then merged upstream. Backported to `3.2.x` as well.
